jps-plugin/src/org/intellij/erlang/jps/builder/ErlangBuilder.java

```diff
@@ -446,84 +378,45 @@ private static void consumeFiles(@NotNull BuildOutputConsumer outputConsumer,
-                         File file,
-                         ErlangSourceRootDescriptor erlangSourceRootDescriptor) throws IOException {
-      ContainerUtil.addIfNotNull(myDirtyElements, getDirtyElement(erlangTarget, file, erlangSourceRootDescriptor));
-      return true;
+  private static void removeOutputFiles(@NotNull Collection<String> removedFiles, @NotNull File... outputDirectories) {
+    for (File dir : outputDirectories) {
+      List<File> outputErlangModuleFiles = ContainerUtil.concat(getBeams(findErlFiles(removedFiles), dir),
+                                                                getAppsOutput(findAppFiles(removedFiles), dir));
+      for (File output : outputErlangModuleFiles) {
+        if (output.exists()) {
+          //noinspection ResultOfMethodCallIgnored
+          output.delete();
+        }
+      }
     }
-
-    @Nullable
-    protected abstract T getDirtyElement(@NotNull ErlangTarget target,
-                                         @NotNull File file,
-                                         @NotNull ErlangSourceRootDescriptor root) throws IOException;
   }
 
-  private static class Node {
-    final String myErlangModulePath;
-    final List<Node> myDependencies = ContainerUtil.newArrayList();
-    public boolean myDirty = false;
-
-    Node(String nodeName) {
-      myErlangModulePath = nodeName;
-    }
+  @NotNull
+  private static List<String> findAppFiles(@NotNull Collection<String> removedFiles) {
+    return ContainerUtil.filter(removedFiles, new Condition<String>() {
+      @Override
+      public boolean value(String filePath) {
+        return isAppConfigFileName(filePath);
+      }
+    });
   }
 
-  private static class SortedModuleDependencyGraph implements GraphGenerator.SemiGraph<Node> {
-    private final LinkedHashMap<String, Node> myNodePathsMap;
-
-    public SortedModuleDependencyGraph(List<ErlangFileDescriptor> moduleDescriptors) {
-      myNodePathsMap = getPathsMap(moduleDescriptors);
-      addDependencies(moduleDescriptors);
-    }
-
-    @Override
-    public Collection<Node> getNodes() {
-      return myNodePathsMap.values();
-    }
-
-    @Override
-    public Iterator<Node> getIn(Node node) {
-      return node.myDependencies.iterator();
-    }
-
-    @Nullable
-    public Node getNode(@NotNull String moduleName) {
-      return myNodePathsMap.get(moduleName);
-    }
-
-    private void addDependencies(List<ErlangFileDescriptor> moduleDescriptors) {
-      for (ErlangFileDescriptor descriptor : moduleDescriptors) {
-        Node node = myNodePathsMap.get(descriptor.myErlangModulePath);
-        node.myDependencies.addAll(getNodesByName(descriptor.myDependencies));
+  private static List<File> getAppsOutput(@NotNull List<String> sourceApps, final File outputDir) {
+    return ContainerUtil.map(sourceApps, new Function<String, File>() {
+      @Override
+      public File fun(String sourceFile) {
+        return new File(outputDir, getAppConfigDestinationFileName(sourceFile));
```

Comment:
This same line is written in the code which copies resource files to output - why not extract a method to get this file?